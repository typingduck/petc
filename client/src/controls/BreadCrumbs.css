
#petc-breadcrumbs {
  border-radius: 10px;
  left: 30%;
  position: absolute;
  top: 30%;
  z-index: var(--z-index-page-bg);
  padding: 10px;
  font-size: larger;
  text-align: center;
  color: rgba(0, 0, 0, 0.2);
  border-color: rgba(0, 0, 0, 0.2);
}
