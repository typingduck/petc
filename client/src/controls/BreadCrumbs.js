import React from 'react'
import './BreadCrumbs.css'

/**
 * A useful development tool for viewing the redux state in the browser.
 * Hidden during normal viewing.
 */
export default function BreadCrumbs (props) {
  let msg = null

  if (props.controls.isNodeMode && Object.keys(props.doc.nodes).length === 0) {
    msg = <p> click on the canvas <br /> to create nodes </p>
  } else if (props.controls.isEdgeMode && Object.keys(props.doc.nodes).length < 2) {
    msg = <p> use the node view <br /> to create 2 or more nodes </p>
  } else if (props.controls.isEdgeMode && Object.keys(props.doc.edges).length === 0) {
    msg = <p> click on nodes <br /> to connect them </p>
  }

  return <div id='petc-breadcrumbs'> {msg} </div>
}
