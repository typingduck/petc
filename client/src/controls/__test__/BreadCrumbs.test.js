/* global expect it */
import React from 'react'
import Renderer from 'react-test-renderer'

import BreadCrumbs from '../BreadCrumbs'

it('renders nothing by default', () => {
  const props = {
    controls: {
      isNodeMode: false,
      isEdgeMode: false
    },
    doc: {
      nodes: {},
      edges: {}
    }
  }
  const component = Renderer.create(<BreadCrumbs {...props} />)
  expect(component.toJSON()).toMatchSnapshot()
})

it('renders nothing for node-view with nodes', () => {
  const props = {
    controls: {
      isNodeMode: true,
      isEdgeMode: false
    },
    doc: {
      nodes: { node: {} },
      edges: {}
    }
  }
  const component = Renderer.create(<BreadCrumbs {...props} />)
  expect(component.toJSON()).toMatchSnapshot()
})

it('renders breadcrumb for edge-view with 1 node', () => {
  const props = {
    controls: {
      isNodeMode: false,
      isEdgeMode: true
    },
    doc: {
      nodes: { node: {} },
      edges: {}
    }
  }
  const component = Renderer.create(<BreadCrumbs {...props} />)
  expect(component.toJSON()).toMatchSnapshot()
})

it('renders breadcrumb for edge-view with 2 nodes no edge', () => {
  const props = {
    controls: {
      isNodeMode: false,
      isEdgeMode: true
    },
    doc: {
      nodes: { node1: {}, node2: {} },
      edges: {}
    }
  }
  const component = Renderer.create(<BreadCrumbs {...props} />)
  expect(component.toJSON()).toMatchSnapshot()
})

it('renders nothing for edge-view with edges', () => {
  const props = {
    controls: {
      isNodeMode: false,
      isEdgeMode: true
    },
    doc: {
      nodes: { node1: {}, node2: {} },
      edges: { edge: {} }
    }
  }
  const component = Renderer.create(<BreadCrumbs {...props} />)
  expect(component.toJSON()).toMatchSnapshot()
})
