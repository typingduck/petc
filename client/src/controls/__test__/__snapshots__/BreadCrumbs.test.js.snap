// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders breadcrumb for edge-view with 1 node 1`] = `
<div
  id="petc-breadcrumbs"
>
   
  <p>
     use the node view 
    <br />
     to create 2 or more nodes 
  </p>
   
</div>
`;

exports[`renders breadcrumb for edge-view with 2 nodes no edge 1`] = `
<div
  id="petc-breadcrumbs"
>
   
  <p>
     click on nodes 
    <br />
     to connect them 
  </p>
   
</div>
`;

exports[`renders nothing by default 1`] = `
<div
  id="petc-breadcrumbs"
>
   
   
</div>
`;

exports[`renders nothing for edge-view with edges 1`] = `
<div
  id="petc-breadcrumbs"
>
   
   
</div>
`;

exports[`renders nothing for node-view with nodes 1`] = `
<div
  id="petc-breadcrumbs"
>
   
   
</div>
`;
